### ISSUE TITLE
[3.4.0] Width and height attributes shouldn't be set if the sizes attribute is specified

### ISSUE THREAD

**Issue by ausi** [contao/core#7500]
We should change the if-condition for the width and height attributes from `!$this->sources` to `!$this->sources && empty($this->img['sizes'])`.

See also <https://community.contao.org/de/showthread.php?54305>

**Comment by aFarkas** [comment]
Removing the ``width``/``height`` attributes is the right thing to do. 

This is maybe a little bit off topic, but you could add a new feature for developers: [**add instrinsic ratio HTML/CSS**](http://alistapart.com/article/creating-intrinsic-ratios-for-video/)
The reason why self-proclaimed SEO experts are demanding to specify width and height attributes, is because they are running page speed insights on a webpage and using these results to consult with.

In case of adaptive images this page speed insight test doesn't make any sense, but the ratio behind this test can accomplished (minimize reflows by reserving/occupying the width and height space for the image) with the so called CSS intrinsic ratio padding-bottom pattern. Unfortunately this means you need to know the exact image dimension ratio, which isn't often known by the frontend. Therefore it would be huge, if developers could opt-in to let contao generate this markup automatically:

```html
<style>
.contao-intrinsic-box {
    display: block;
    position: relative;
    height: 0;
    overflow: hidden;
}
.contao-intrinsic-box img,
.contao-intrinsic-box video,
.contao-intrinsic-box iframe,
.contao-intrinsic-box object {
    display: block;
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}
</style>


<span class="contao-intrinsic-box contao-landscape-box" style="padding-bottom: 42.86%">
    <img
        src="http://placehold.it/175x75"
        sizes="foobar"
        srcset="http://placehold.it/175x75 175w,
        http://placehold.it/350x150 350w,
        http://placehold.it/700x300 700w,
        http://placehold.it/1400x600 1400w" />
</span>
```

This would be really huge. There are also a lot of scripts out there (mostly carousels/sliders, but also isotope/masonry) trying to wait until all images are loaded to get the right dimension of their items. These scripts currently epically fail as soon as it comes to responsive images. With the intrinsic ratio pattern those problems are also automatically solved. What do you think? Should I open a new issue for this feature?

Additionally you can also think of adding the height descriptor. While this descriptor isn't in use by any browser yet, there is some discussion to use this to also solve this problem regardless of the full height descriptor implementation:

```html
<img
        src="http://placehold.it/175x75"
        sizes="foobar"
        srcset="http://placehold.it/175x75 175w 75h,
        http://placehold.it/350x150 150h 350w,
        http://placehold.it/700x300 300h 700w,
        http://placehold.it/1400x600 600h 1400w" />
```

**Comment by ausi** [comment]
@aFarkas thanks for your input.

I think this would be a great feature but it’s also a bit complicated. I’m not sure if this should be in the core or in an external extension.

Do you have more information about the height descriptor? Is it backwards compatible with current browser implementations?

> Should I open a new issue for this feature?

Yes please, because your idea is a feature and this issue is about a bug which will be fixed in the next patch release.

**Comment by leofeyer** [comment]
Fixed in f9995c96c649f2435c4d73c74fd09911936071b4.

@ausi Can you please confirm that my patch works? Instead of using

```php
if (!$this->sources && empty($this->img['sizes'])):
```

I have combined it with the previous if statement:

```php
<?php if (!empty($this->img['sizes'])): ?> sizes="<?php echo $this->img['sizes']; ?>"<?php elseif (!$this->sources): ?> width="<?php echo $this->img['width']; ?>" height="<?php echo $this->img['height']; ?>"<?php endif; ?>
```

**Comment by ausi** [comment]
@leofeyer confirmed.
